tests/v3_api/test_logging.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+import requests
+import time
+from .common import random_test_name
+from .common import get_admin_client_and_cluster
+from .common import create_project_and_ns
+from .common import get_project_client_for_token
+from .common import create_kubeconfig
+from .common import get_admin_client
+from .common import CATTLE_TEST_URL
+from .common import ADMIN_TOKEN
+
+namespace = {"p_client": None, "ns": None, "cluster": None, "project": None,
+             "name_prefix": None}
+answers = {
+        "defaultImage": "true",
+        "image": "confluentinc/cp-kafka",
+        "imageTag": "4.0.1-1",
+        "zookeeper.image.repository": "confluentinc/cp-zookeeper",
+        "zookeeper.image.tag": "4.1.1",
+        "schema-registry.enabled": "false",
+        "kafka-rest.enabled": "false",
+        "replicas": "3",
+        "persistence.enabled": "false",
+        "zookeeper.persistence.enabled": "false",
+        "kafka-topics-ui.enabled": "false",
+        "kafka-topics-ui.ingress.enabled": "false",
+    }
+
+CATTLE_ClUSTER_LOGGING_KAFKA_TEST = \
+    CATTLE_TEST_URL + "/v3/clusterloggings?action=test"
+CATTLE_PROJECT_LOGGING_KAFKA_TEST = \
+    CATTLE_TEST_URL + "/v3/projectloggings?action=test"
+KAFKA_CATALOG_ID = "catalog://?catalog=library&template=kafka&version=0.7.3"
+
+
+def test_send_log_to_kafka(setup_app):
+    cluster = namespace["cluster"]
+    project = namespace["project"]
+    send_log_to_kafka(CATTLE_ClUSTER_LOGGING_KAFKA_TEST, cluster.id,
+                      project.id, ADMIN_TOKEN)
+    send_log_to_kafka(CATTLE_PROJECT_LOGGING_KAFKA_TEST, cluster.id,
+                      project.id, ADMIN_TOKEN)
+
+
+def send_log_to_kafka(url, clusterId, projectId, token, expected_status=204):
+    headers = {'Authorization': 'Bearer ' + token}
+    endpoint = "http://" + namespace["name_prefix"] + "-kafka" + "." + \
+        namespace["ns"].name + ".svc.cluster.local:9092"
+    print("kafka endpoint:"+endpoint)
+    kafkaConfig = {
+        "brokerEndpoints": [endpoint],
```

Comment:
It would be nice to have "unhappy path" tests to check that the API handles invalid endpoints input properly.